packages/frontend/src/contexts/game/reducer/index.js

```diff
@@ -37,28 +39,16 @@ export const gameReducer = (state, action) => {
       return tempState;
     }
 
-    case types.RESET_TILES:
+    case types.HANDLE_FLIPPED_PAIR:
       return {
         ...state,
-        tiles: state.tiles.map((tile) =>
-          action.payload.tiles[0].id === tile.id ||
-          action.payload.tiles[1].id === tile.id
-            ? { ...tile, faceUp: !tile.faceUp, isFlippable: true }
-            : { ...tile, isFlippable: true }
-        ),
-        flipped: [],
-      };
-
-    case types.HANDLE_MATCH:
-      return {
-        ...state,
-        tiles: state.tiles.map((tile) =>
-          action.payload.tiles[0].id === tile.id ||
-          action.payload.tiles[1].id === tile.id
-            ? { ...tile, isMatched: !tile.isMatched }
-            : { ...tile, isFlippable: true }
-        ),
+        tiles:
+          action.payload.tiles[0].photo.id === action.payload.tiles[1].photo.id
```

Comment:
There could be another helper function like `isMatchingPair` or something that could encapsulate this if statement. `tiles: isMatchingPair(tile1, tile2) ? handleMatch(state.tiles, action.payload.tiles) : resetTiles(state.tiles)`